### PR TITLE
🌱Add test with CAPI E2E testing framework

### LINKED ISSUE #191: Add e2e tests consuming CAPI's e2e testing framework
**User Story**
Currently, we do not have E2E tests in the repo and we would like to consume [CAPI's e2e testing framework](https://pkg.go.dev/sigs.k8s.io/cluster-api/test/framework) to make this provider similar to the others and easy to maintain the tests. 

As an operator we would like to:

- Provide consistent e2e expectations across infrastructure providers
- Have a testing framework to test all upcoming new features into the project
- Utilize common end-to-end patterns that can be reused across Cluster API providers

**Anything else you would like to add:**
Relative links to get started:

- [Cluster API E2E walkthrough shared doc](https://docs.google.com/document/d/1MhpZm6ZfjxDtqZlwo3BR4XyqtS3XBTVFUZ8-OL_ZXjA/edit#heading=h.w4ycat3brb3e)
- [CAPI book guide to develop e2e tests](https://cluster-api.sigs.k8s.io/developer/e2e.html)
- [Initial discussion issue started in CAPI](https://github.com/kubernetes-sigs/cluster-api/issues/4406)

/kind feature

### PR DESCRIPTION
**Pull request by Insullone**
Add test in HA cluster with CAPI E2E testing framework

Fixes #191

### CODE REVIEW

File: test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+	kubeconfigPath string
+	e2eTestsPath string
+)
+
+func init() {
+	flag.StringVar(&configPath, "e2e.config", "", "path to the e2e config file")
+	flag.StringVar(&artifactFolder, "e2e.artifacts-folder", "", "folder where e2e test artifact should be stored")
+	flag.BoolVar(&skipCleanup, "e2e.skip-resource-cleanup", false, "if true, the resource cleanup after tests will be skipped")
+	flag.BoolVar(&useExistingCluster, "e2e.use-existing-cluster", false, "if true, the test uses the current cluster instead of creating a new one (default discovery rules apply)")
```
[the remark this answers is not in the document]
Comment (Insullone) [reply test/e2e/e2e_suite_test.go:67]: OK

File: test/e2e/e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+					KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
+					InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+					Flavor:                   "ha",
+					Namespace:                namespace,
+					ClusterName:              clusterName,
+					KubernetesVersion:        e2eConfig.GetVariable(KubernetesVersion),
+					ControlPlaneMachineCount: pointer.Int64Ptr(3),
+					WorkerMachineCount:       pointer.Int64Ptr(1),
+				},
+				CNIManifestPath:              e2eTestsPath + "/data/cni/calico/calico.yaml",
```
Comment (namnx228) [inline test/e2e/e2e_test.go:59]: Should we use the calico.yaml locally, or should we refer to a URL? For example, https://docs.projectcalico.org/v3.15/manifests/calico.yaml

**Review by namnx228 — COMMENTED** [review]
Impressive work. Look good to me

File: test/e2e/e2e_test.go (hunk 2 above)
Comment (furkatgofurov7) [reply test/e2e/e2e_test.go:59]: @namnx228 IMO using it locally is fine, as it is used in other provider repos too.

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Review by kashifest — APPROVED** [review]
Two minor comments:
1. We can use local registry images for controllers
2.  Parameterization of the variables would be expected in some future work.

**Pull request merged**